zb-classifier.js

```diff
@@ -83,22 +83,138 @@ const ZONE_TYPE_NAME = {
   // non-switch sensors which also report a zoneType of 0, we may need
   // to use further refinements, like the presence of the binaryInput
   // cluster.
-  0x0000: {name: 'switch', label: 'Open', descr: 'Contact Switch'},
-  0x000d: {name: 'motion', label: 'Motion', descr: 'Motion Sensor'},
-  0x0015: {name: 'switch', label: 'Open', descr: 'Contact Switch'},
-  0x0028: {name: 'fire', label: 'Fire', descr: 'Fire Sensor'},
-  0x002a: {name: 'water', label: 'Water', descr: 'Water Sensor'},
-  0x002b: {name: 'co', label: 'CO', descr: 'Carbon Monoxide Sensor'},
-  0x002c: {name: 'ped', label: 'Pressed', descr: 'Personal Emergency Device'},
+  0x0000: {
+    name: 'switch',
+    '@type': ['DoorSensor'],
+    propertyName: 'open',
+    propertyDescr: {
+      '@type': 'OpenProperty',
+      type: 'boolean',
+      label: 'Open',
+      descr: 'Contact Switch',
+    },
+  },
+  0x000d: {
+    name: 'motion',
+    '@type': ['MotionSensor'],
+    propertyName: 'motion',
+    propertyDescr: {
+      '@type': 'MotionProperty',
+      type: 'boolean',
+      label: 'Motion',
+      descr: 'Motion Sensor',
+    },
+  },
+  0x0015: {
+    name: 'switch',
+    '@type': ['DoorSensor'],
+    propertyName: 'open',
+    propertyDescr: {
+      '@type': 'OpenProperty',
+      type: 'boolean',
+      label: 'Open',
+      descr: 'Contact Switch',
+    },
+  },
+  0x0028: {
+    name: 'fire',
+    '@type': ['BinarySensor'],
+    propertyName: 'on',
```

Comment:
Most of these should not be named "on", otherwise they may end up conflicting.

Reply:
There can only be one of these for a given sensor (since there is only one zoneType per sensor).